File: 2302-count-subarrays-with-score-less-than-k/2302-count-subarrays-with-score-less-than-k.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        i=0
        j=0
        summ=0
        score=0
        count=0
        while(j<len(nums)):
            summ+=nums[j]
            score=summ*(j-i+1)
            while(score>=k and i<=j):
                summ-=nums[i]
                i+=1
                score=summ*(j-i+1)
            count+=(j-i+1)
            j+=1
        return count

File: 2302-count-subarrays-with-score-less-than-k/2302-count-subarrays-with-score-less-than-k.c
long long countSubarrays(int* nums, int numsSize, long long k) {
        int i=0;
        int j=0;
        long long sum=0;
        long long score=0;
        long long count=0;
        while(j<numsSize)
        {
            sum+=nums[j];
            score=sum*(j-i+1);
            while(score>=k && i<=j)
            {
                sum-=nums[i];
                i++;
                score=sum*(j-i+1);
            }
            count+=(j-i+1);
            j++;
        }
        return count;
}